packages/dynamite/dynamite/lib/src/builder/generate_ofs.dart

```diff
@@ -0,0 +1,243 @@
+    yield TypeDef((final b) {
+      b
+        ..name = result.name
+        ..definition = refer(result.dartType.name);
+    });
+
+    if (result.optimizedSubTypes.length <= 1) {
+      continue;
+    }
+    yield* generateSomeOf(result);
+  }
+}
+
+Iterable<Spec> generateSomeOf(
+  final TypeResultSomeOf result,
+) sync* {
+  final identifier = result.name;
+  final results = result.optimizedSubTypes;
+  final serializerName = '_\$${identifier}Serializer';
+
+  final fields = <TypeResult, String>{};
+  for (final result in results) {
+    final dartType = result.nullableName;
+    final dartName = toDartName(dartType);
+    fields[result] = toFieldName(dartName, result.className);
+  }
+
+  final values = Method((final b) {
+    b
+      ..returns = refer('List<dynamic>')
+      ..type = MethodType.getter
+      ..name = '_values'
+      ..lambda = true
+      ..body = Code('[${fields.values.join(',')}]');
+  });
+
+  final oneOfValidator = Method((final b) {
+    b
+      ..name = 'validateOneOf'
+      ..returns = refer('void')
+      ..lambda = true
+      ..body = const Code('dynamite_utils.validateOneOf(_values)');
+  });
+
+  final anyOfValidator = Method((final b) {
+    b
+      ..name = 'validateAnyOf'
+      ..returns = refer('void')
+      ..lambda = true
+      ..body = const Code('dynamite_utils.validateAnyOf(_values)');
+  });
+
+  final serializerMethod = Method(
+    (final b) => b
+      ..static = true
+      ..returns = refer('Serializer<$identifier>')
+      ..type = MethodType.getter
+      ..name = 'serializer'
+      ..lambda = true
+      ..body = Code('const $serializerName()'),
+  );
+
+  final fromJson = Method(
+    (final b) => b
+      ..static = true
+      ..returns = refer(identifier)
+      ..name = 'fromJson'
+      ..requiredParameters.add(
+        Parameter(
+          (final b) => b
+            ..name = 'json'
+            ..type = refer('Object?'),
+        ),
+      )
+      ..lambda = true
+      ..body = const Code('_jsonSerializers.deserializeWith(serializer, json)!'),
+  );
+
+  final toJson = Method(
+    (final b) => b
+      ..name = 'toJson'
+      ..returns = refer('Object?')
+      ..lambda = true
+      ..body = const Code('_jsonSerializers.serializeWith(serializer, this)'),
+  );
+
+  yield Extension(
+    (final b) => b
+      ..name = '${identifier}Extension'
+      ..on = refer(identifier)
+      ..methods.addAll([
+        values,
+        oneOfValidator,
+        anyOfValidator,
+        serializerMethod,
+        fromJson,
+        toJson,
+      ]),
+  );
+
+  yield Class(
+    (final b) => b
+      ..name = serializerName
+      ..implements.add(refer('PrimitiveSerializer<$identifier>'))
+      ..constructors.add(Constructor((final b) => b.constant = true))
+      ..methods.addAll([
+        Method(
+          (final b) => b
+            ..annotations.add(refer('override'))
+            ..returns = refer('Iterable<Type>')
+            ..type = MethodType.getter
```

Comment:
Can just be a const field

Reply:
Nope. `avoid_field_initializers_in_const_classes`
https://dart.dev/tools/linter-rules/avoid_field_initializers_in_const_classes